runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/io/SdkSource.kt

```diff
@@ -43,16 +51,42 @@ public interface SdkSource : Closeable {
  * Consume the [SdkSource] and pull the entire contents into memory as a [ByteArray].
  */
 @InternalApi
-public expect suspend fun SdkSource.readToByteArray(): ByteArray
+public suspend fun SdkSource.readToByteArray(): ByteArray = withContext(SdkDispatchers.IO) {
+    use { it.buffer().readByteArray() }
+}
```

Comment:
note: the only reason I could see these were `expect/actual` is because of `Dispatchers.IO`, but we already have an `expect/actual` version of that called `SdkDispatchers`, so I've moved these to **common** and deleted the JVM/Native implementations